### ISSUE TITLE
Better way to manage group for entries

### ISSUE THREAD

**Issue by ror3d** [JabRef/jabref#11026]
**Is your suggestion for improvement related to a problem? Please describe.**
Currently, as far as I'm aware, the only way to remove an entry from a group is by selecting the entry (or a set of entries), right clicking on a group, and checking "remove selected entries from this group".

That is a bit cumbersome, but specially it's not intuitive. The group system is basically a tag/category system (since an entry can belong to many groups at once) and most systems that use tags have a way to manage the tags an entry belongs to from the entry itself, not from the tags.

**Describe the solution you'd like**
Either a right click on the entry > groups > checking/unchecking the ones it should belong to, or some other way that a list with checkboxes (or similar) can be accessed to manage them, such as a dropdown somewhere that opens the list of groups with checkboxes that that entry has assigned. This option would allow for editing the groups of multiple entries at once.

Another option would be to have these tags also accessible from some tab in the entry editor panel, although that would only allow for one entry at a time to be modified.

**Additional context**
For example, outlook's solution, 
![image](https://github.com/JabRef/jabref/assets/1844478/d4bbfbf4-1993-4d03-8572-c54c47d278f7)
(this one requires right-clicking for every category you want to add or remove, which is not ideal, much better would be that you could select and unselect several with opening it only once)

**Comment by koppor** [comment]
We somewhere have discussion to do following:

- groups should be tags (which can be categorized)
- new "taxonomy" tree - similar to folders (https://github.com/JabRef/jabref/issues/10930)

The idea is "copied" from TagSpaces: https://www.tagspaces.org/

**Comment by koppor** [comment]
I need input on following: We had en exhausting debate on the "keywords" field. One opinion: it is from the publisher AND MOST NOT be changed. Other opinion: This is what I categorize entries with - and it should be automatically reflected in the groups ("keyword groups" is one step). -- Maybe the "solution" is to have a "tags" field. `groups` would be JabRef 5.x behavio, and `tags` a new JabRef 6.x behavior.

(Side note: We changed the group format in the 3.x line inbetween and that caused confusion -- https://github.com/JabRef/jabref/blob/v4.0/CHANGELOG.md#34--2016-06-02).

**Comment by koppor** [comment]
Just as note to think of it later.

`dret.bib` (https://github.com/dret/biblio) has the idea of `topics`:

```bibtex
topic =     "jsp[0.7] asp[0.7] perl[0.7] cgi[0.7] servlet[0.7] php[0.7]",
```

It seems that these are "tags" with a matching ratio added.

(Refs https://github.com/JabRef/jabref-issue-melting-pot/issues/421)

**Comment by koppor** [comment]
Another note: JabRef supports hierarchical keywords (`a > b > c`). Maybe, we could use that for tags (or taxonomy items), too?

**Comment by koppor** [comment]
The feature of hiearchical keywords is explained at https://github.com/JabRef/jabref/issues/628

**Comment by andr-groth** [comment]
Since the new keyword tag systems #10910  does not supporty hierarchical keywords anymore, I'd would be happy to see this feature #628  restored. This could be hopefully added to the tag systems, by extending the tag box:

![Untitled Diagram](https://github.com/JabRef/jabref/assets/94175647/51cb3281-5826-42b1-8ae5-2f415a7d2751)

On the other hand, the feature #628 is already quite powerful:

- hierarchical groups can be automatically generated from keywords
- entries can be assigned by drag&drop,
- adding new entries will be automatically assigned if one or more groups are selected,
- publisher keywords are merged with the user defined keywords and create new groups,
- ...

But since keywords are not required, they do not immediatly appear on the _required_ tab when editing an entry. This is maybe why the feature #628 is not very well known. Only mentioned in a short paragraph of the documentation on groups (even not mentioned in the keywords section) : https://docs.jabref.org/finding-sorting-and-cleaning-entries/groups#nesting-sub-groups

## Suggestion

What I would suggest is to make this feature #628 more visible, in the documentation and in the UI. The keyword tags are a good starting point :thumbsup: and maybe they should be made more prominent in the UI:

1. Explain  hierarchical keywords in https://docs.jabref.org/finding-sorting-and-cleaning-entries/keywords
2. The keyword field could be made sticky and appear below the required fields.
3. Keyword tags could appear on top of the entry preview.
4. The _general_ tab could be made more prominent.

On the other hand, adding a new _tag_ field might add a new level of complexity and confuse the user. I would keep it with the Occam's razor and the principle of parsimony.

**Comment by andr-groth** [comment]
And of course I would second the long-term goal #6191. At the moment there are over 30 input fields spread across six different tabs.

**Comment by andr-groth** [comment]
> And of course I would second the long-term goal #6191. At the moment there are over 30 input fields spread across six different tabs.

Please see also my remark on koppor/jabref#679.

**Comment by LoayGhreeb** [comment]
> However they're not folders because I can have an entry in a child group, but it can not belong to the parent group. That's the thing I got very-very confused with.

If you set the parent group's hierarchy to union, it will display all entries within both the parent and child groups. However, this won't affect the "groups" field in the bib source.

**Comment by ThiloteE** [comment]
You are right Ruslan, JabRef's groups feature is actually a tags feature, but we call it groups :D

IMHO, the larger your library becomes, the more difficult it is to manage via folders (because it takes long to find entries in deep nested structures; Too much scrolling, too much clicking). Folders are good for small and mid sized libraries, whereas large libraries work better with tags.

Both tagss and folders profit immensely from a workflow that revolves around "search" or "search groups", but tags especially, because they easily allow you to find entries that haven't been categorized into groups yet. Categorizing via groups is a second more labor intensive step, whereas many entries already come with existing tags and keywords so less work is required.

**Comment by InAnYan** [comment]
Okay, I see.

After some thinking, I came to conclusion, that it is faster to make a keyword group for "document management" or "Ukraine", rather than manually moving them...

Then okay. I don't see any problems of changing groups to tags.

But, I have questions:

1. Can tags be nested?
2. Can tags be assigned both manually and automatically?
